### PR TITLE
fix(display): fix inconsistency in the API

### PR DESCRIPTION
**Pull request by kisvegabor**
### Description of the feature or fix

A clear and concise description of what the bug or new feature is.

### Checkpoints
- [ ] Run `code-format.py` from the scripts folder. [astyle](http://astyle.sourceforge.net/install.html) needs to be installed.
- [ ] Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed
- [ ] Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant. 
- [ ] Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- [ ] If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/release/v8.3/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/release/v8.3/Kconfig).
 
Be sure the following conventions are followed:
- [ ] Follow the [Styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [ ] Prefer `enum`s instead of macros. If inevitable to use `define`s export them with `LV_EXPORT_CONST_INT(defined_value)` right after the `define`.
- [ ] In function arguments prefer `type name[]` declaration for array parameters instead of `type * name`
- [ ] Use typed pointers instead of `void *` pointers
- [ ] Do not `malloc` into a static or global variables. Instead declare the variable in `lv_global_t` structure in [`lv_global.h`](https://github.com/lvgl/lvgl/blob/master/src/core/lv_global.h) and mark the variable with `(LV_GLOBAL_DEFAULT()->variable)` when it's used. See a detailed description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#memory-management).
- [ ] Widget constructor must follow the `lv_<widget_name>_create(lv_obj_t * parent)` pattern.
- [ ] Widget members function must start with `lv_<module_name>` and should receive `lv_obj_t *` as first argument which is a pointer to widget object itself.  
- [ ] `struct`s should be used via an API and not modified directly via their elements.
- [ ] `struct` APIs should follow the widgets' conventions. That is to receive a pointer to the `struct` as the first argument, and the prefix of the `struct` name should be used as the prefix of the function name too (e.g.  `lv_disp_set_default(lv_disp_t * disp)`)
- [ ] Functions and `struct`s which are not part of the public API must begin with underscore in order to mark them as "private".
- [ ] Arguments must be named in H files too.
- [ ] To register and use callbacks one of the following needs to be followed (see a detailed description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#callbacks)):
  - For both the registration function and the callback pass a pointer to a `struct` as the first argument. The `struct` must contain `void * user_data` field.
  - The last argument of the registration function must be `void * user_data` and the same `user_data` needs to be passed as the last argument of the callback.
  - Callback types not following these conventions should end with `xcb_t`.

### CODE REVIEW

**Comment by kisvegabor** [comment]
@kdschlosser, please take a look.

**Comment by kdschlosser** [comment]
it looks good... I haven't tested it at all yet. I do have a question tho. does LVGL support multiple displays being attached at one time?

**Comment by kisvegabor** [comment]
> does LVGL support multiple displays being attached at one time?

Yes, see [here](https://docs.lvgl.io/master/overview/disp.html#multiple-display-support).

**Comment by kdschlosser** [comment]
If there is multi display support then this code is not right

```
void lv_display_load_screen(lv_display_t * disp, lv_obj_t * scr)
{
    LV_UNUSED(disp); /*Not required, added just for consistency*/
    lv_screen_load_anim(scr, LV_SCR_LOAD_ANIM_NONE, 0, 0, false);
}
```

Because you are not loading a screen to a specific display you are loading it to whatever display is currently being used. adding the display as a parameter only for the purposes of the API should not be done. The screen should actually get added to the display that was passed into it. There should be a requirement for the display to be valid. the lv_screen_load should locate the display that is currently in use and pass that when it makes the call to lv_display_load_screen.

**Comment by kisvegabor** [comment]
The screen is already created on a given display. So from the screen widget LVGL get the related display, e.g. with `lv_obj_get_disp` (I'm updating it to `lv_obj_get_display`)

**Review by FASTSHIFT — DISMISSED** [review]

**Comment by kdschlosser** [comment]
OK so when I do the following..

```c
lv_obj_t *new_screen = lv_create_obj(NULL);
lv_display_load_screen(some_display, new_screen);
```

What display does the new screen get attached to? If I have more than a single display `some_display` might not be the one that is active. but LVGL is making the assumption that when I want to load a screen that means that I want to see it on the active display and that might not be the case.

see what I am getting at now?

File: src/display/lv_display.c [hunk 1]
```diff
@@ -492,8 +492,9 @@ lv_obj_t * lv_display_get_screen_prev(lv_display_t * disp)
     return disp->prev_scr;
 }
 
-void lv_display_load_scr(lv_obj_t * scr)
+void lv_display_load_screen(lv_display_t * disp, lv_obj_t * scr)
```
Comment (XuNeo) [inline src/display/lv_display.c]: Need to add to `lv_api_map.h`

File: src/core/lv_obj_pos.c [hunk 2]
```diff
@@ -282,7 +282,7 @@ void lv_obj_mark_layout_as_dirty(lv_obj_t * obj)
     scr->scr_layout_inv = 1;
 
     /*Make the display refreshing*/
-    lv_display_t * disp = lv_obj_get_disp(scr);
+    lv_display_t * disp = lv_obj_get_display(scr);
```
Comment (XuNeo) [inline src/core/lv_obj_pos.c:285]: Need add `lv_obj_get_display ` to api map.

File: src/indev/lv_indev.h [hunk 3]
```diff
@@ -137,9 +137,9 @@ lv_indev_state_t lv_indev_get_state(const lv_indev_t * indev);
 
 lv_group_t * lv_indev_get_group(const lv_indev_t * indev);
 
-struct _lv_display_t * lv_indev_get_disp(const lv_indev_t * indev);
+struct _lv_display_t * lv_indev_get_display(const lv_indev_t * indev);
 
-void lv_indev_set_disp(lv_indev_t * indev, struct _lv_display_t * disp);
+void lv_indev_set_display(lv_indev_t * indev, struct _lv_display_t * disp);
```
Comment (XuNeo) [inline src/indev/lv_indev.h:134]: Add `lv_indev_get_display` to api map.

File: src/display/lv_display.c (hunk 1 above)
Comment (kisvegabor) [reply src/display/lv_display.c]: Finally I removed this function as loading a screen is not related to the display. `lv_screen_load()` can be used directly.

File: src/core/lv_obj_pos.c (hunk 2 above)
Comment (kisvegabor) [reply src/core/lv_obj_pos.c:285]: Added

File: src/indev/lv_indev.h (hunk 3 above)
Comment (kisvegabor) [reply src/indev/lv_indev.h:134]: Added

**Comment by kisvegabor** [comment]
> What display does the new screen get attached to?

`lv_obj_create(NULL)` will create a screen on the current default display. Which is the first display the your create or what you set with `lv_display_set_default`. There is no way to cross-load screens among displays. A screen belongs to the display where your have created it. 

To resolve this inconsistency I removed `lv_display_load_screen` and users can use `lv_screen_load` directly.

**Review by XuNeo — APPROVED** [review]
Need to solve the conflicts firstly.

**Review by FASTSHIFT — DISMISSED** [review]

**Comment by kisvegabor** [comment]
> Need to solve the conflicts firstly.

Resolved

File: src/display/lv_display.h [hunk 4]
```diff
@@ -329,6 +329,15 @@ lv_obj_t * lv_display_get_screen_active(lv_display_t * disp);
...
 /**
+<<<<<<< HEAD
+=======
+ * Make a screen active
+ * @param scr       pointer to a screen
+ */
+void lv_display_load_scr(lv_obj_t * scr);
+
+/**
+>>>>>>> master
```
Comment (XuNeo) [inline src/display/lv_display.h]:
```suggestion
```

**Review by PGNetHun — APPROVED** [review]

**Review by XuNeo — APPROVED** [review]

**Review by FASTSHIFT — APPROVED** [review]

**Review by FASTSHIFT — APPROVED** [review]

**Pull request merged**